test/functional/test_framework/key.py

```diff
@@ -444,3 +519,68 @@ def tweak_add(self, tweak):
         ret = ECKey()
         ret.set(tweaked.to_bytes(32, 'big'), self.compressed)
         return ret
+
+def generate_schnorr_nonce():
+    """Generate a random valid Schnorr nonce."""
```

Comment:
This should be "valid bip-schnorr nonce" rather than "valid Schnorr nonce", and should explain what makes it valid (that it is a quadratic residue). You could also refer to the signing algorithm in bip-schnorr (https://github.com/sipa/bips/blob/bip-schnorr/bip-schnorr.mediawiki#Signing) and say that this isn't the deterministic algorithm used there.